webdriver/tests/bidi/script/call_function/result_ownership.py

```diff
@@ -0,0 +1,81 @@
+                         [("root", True), ("none", False), (None, False)])
+async def test_invalid_script(bidi_session, top_context, result_ownership, should_contain_handle):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.call_function(
+            function_declaration="))) !!@@## some invalid JS script (((",
+            await_promise=False,
+            result_ownership=result_ownership,
+            target=ContextTarget(top_context["context"]))
+
+    if should_contain_handle:
+        assert "handle" in exception.value.result["exceptionDetails"]["exception"], \
+            f"Exception should contain `handle`. Actual: {exception.value.result['exceptionDetails']['exception']}"
+    else:
+        assert "handle" not in exception.value.result["exceptionDetails"]["exception"], \
+            f"Exception should not contain `handle`. Actual: {exception.value.result['exceptionDetails']['exception']}"
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("result_ownership, should_contain_handle",
+                         [("root", True), ("none", False), (None, False)])
+async def test_rejected_promise(bidi_session, top_context, result_ownership, should_contain_handle):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.call_function(
+            function_declaration="()=>{return Promise.reject({a:1})}",
+            await_promise=True,
+            result_ownership=result_ownership,
+            target=ContextTarget(top_context["context"]))
+
+    if should_contain_handle:
+        assert "handle" in exception.value.result["exceptionDetails"]["exception"], \
+            f"Exception should contain `handle`. Actual: {exception.value.result['exceptionDetails']['exception']}"
+    else:
+        assert "handle" not in exception.value.result["exceptionDetails"]["exception"], \
+            f"Exception should not contain `handle`. Actual: {exception.value.result['exceptionDetails']['exception']}"
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("await_promise", [True, False])
+@pytest.mark.parametrize("result_ownership, should_contain_handle",
+                         [("root", True), ("none", False), (None, False)])
+async def test_return_value(bidi_session, top_context, await_promise, result_ownership, should_contain_handle):
+    result = await bidi_session.script.call_function(
+        function_declaration="async function(){return {a:1}}",
+        await_promise=await_promise,
+        result_ownership=result_ownership,
+        target=ContextTarget(top_context["context"]))
+
+    if should_contain_handle:
+        assert "handle" in result, \
+            f"Result should contain `handle`. Actual: {exception.value.result['exceptionDetails']['exception']}"
+    else:
+        assert "handle" not in result, \
+            f"Result should not contain `handle`. Actual: {exception.value.result['exceptionDetails']['exception']}"
```

Comment:
exception is not available here
```suggestion
    if should_contain_handle:
        assert "handle" in result, \
            f"Result should contain `handle`. Actual: {result}"
    else:
        assert "handle" not in result, \
            f"Result should not contain `handle`. Actual: {result}"
```